File: flo-clear-or-snooze-migration/src/constants/SystemMode.ts
import _ from 'lodash';

const SystemMode = {
  HOME: {
    id: 2,
    name: "home"
  },
  AWAY: {
    id: 3,
    name: "away"
  },
  SLEEP: {
    id: 5,
    name: "sleep"
  },
  UNKNOWN: {
    id: -1,
    name: "unknown"
  }
};

function fromName(name: string) {
  return _.values( _.pickBy(SystemMode, (value, _) => {
    return value.name == name;
  }))[0];
}

function fromId(id: number) {
  return _.values( _.pickBy(SystemMode, (value, _) => {
    return value.id == id;
  }))[0];
}

export default {
  ...SystemMode,
  fromName,
  fromId
};

File: flo-android-app/lib/model/system_mode.dart
library system_mode;

import 'package:built_collection/built_collection.dart';
import 'package:built_value/built_value.dart';
import 'package:built_value/serializer.dart';

part 'system_mode.g.dart';

@BuiltValueEnum(wireName: 'systemMode')
class SystemMode extends EnumClass {
  
  static Serializer<SystemMode> get serializer => _$systemModeSerializer;

  @BuiltValueEnumConst(wireName: 'home')
  static const SystemMode home = _$wireHome;

  @BuiltValueEnumConst(wireName: 'away')
  static const SystemMode away = _$wireAway;

  @BuiltValueEnumConst(wireName: 'sleep')
  static const SystemMode sleep = _$wireSleep;

  const SystemMode._(String name) : super(name);

  static BuiltSet<SystemMode> get values => _$wireValues;
  static SystemMode valueOf(String name) => _$wireValueOf(name);

  static const HOME = "home";
  static const SLEEP = "sleep";
  static const AWAY = "away";
}